plugins/get_url/komand_get_url/triggers/poll_file/trigger.py

```diff
@@ -14,65 +16,47 @@ def __init__(self):
         )
+        self.is_modified = True
+        self.utils = Utils(action=self)
 
     def run(self, params={}):
-        utils = Utils(action=self)
+        poll = params.get(Input.POLL, constants.DEFAULT_TIMEOUT)
         url = params.get(Input.URL)
         is_verify = params.get(Input.IS_VERIFY, True)
-        poll = params.get(Input.POLL, 60)
-
+        user_agent = params.get(Input.USER_AGENT, constants.DEFAULT_USER_AGENT)
         while True:
-            """Check for supported url prefix"""
-            utils.validate_url(url)
-
-            is_modified = True
-            meta = utils.hash_url(url)
-            cache_file = "/var/cache/" + meta["file"]
-
-            """Attempt to retrieve headers from past request"""
-            headers = {}
-            if komand.helper.check_cachefile(meta["metafile"]):
-                headers = utils.check_url_meta_file(meta)
-
-            """Download file"""
-            url_object = komand.helper.open_url(
-                url,
-                verify=is_verify,
-                If_None_Match=headers.get("etag", ""),
-                If_Modified_Since=headers.get("last-modified", ""),
-                User_Agent=params.get(Input.USER_AGENT, "Mozilla/5.0"),
-            )
-
-            """File modified"""
+            url_object, meta = self.utils.check_prefix_and_download(url, is_verify, user_agent)
+            # File modified
             if url_object:
-                contents = url_object.read().decode("utf-8")
-
-                """Write etag and last modified to cache"""
-                utils.create_url_meta_file(meta, url_object)
-
-                """We can't guarantee server supports lastmodified/etag, compare contents"""
-                if komand.helper.check_cachefile(cache_file):
-                    old = komand.helper.open_cachefile(cache_file)
-                    old_contents = old.read()
-                    old.close()
-                    if old_contents == contents:
-                        is_modified = False
-                        self.logger.info("GetUrl: File not updated")
-
-                """Write new URL file contents to cache"""
-                if is_modified:
-                    f = komand.helper.open_cachefile(cache_file)
-                    f.write(contents)
-                    f.close()
-
-                    """Check URL status code and return file contents"""
-                    if url_object.code >= 200 or url_object.code <= 299:
-                        f = komand.helper.encode_string(contents)
-                        if f:
-                            output = {
-                                Output.BYTES: f.decode("utf-8"),
-                                Output.STATUS_CODE: url_object.code or 200,
-                            }
-                            self.send(output)
-
+                self._save_to_cache_and_send(url_object, meta)
             time.sleep(poll)
+
+    def _save_to_cache_and_send(self, url_object, meta):
```

Comment:
A docstring would be a bliss

Reply:
We try to stick to clean code rules. The code already has comments that additionally describe what is happening in the method. I think the code itself describes well what the function does. If you don't understand something, please give me information what is not clear and I'll try to make code even more readable.

Reply:
@akarolik-r7 Kindly add the comments on what the method is doing.

Reply:
It is clean its true. However, documenting at least private methods should be beneficial for the code quality in the long run.

Reply:
Additionally, can you please type-hint the function arguments and return?

Reply:
I have add doc string for function. And type-hint for all functions in utils